cpp/src/arrow/array-binary-test.cc

```diff
@@ -42,6 +42,30 @@ using internal::checked_cast;
 // ----------------------------------------------------------------------
 // String / Binary tests
 
+template <typename ArrayType>
+void CheckStringArray(const ArrayType& array, const std::vector<std::string>& strings,
+                      const std::vector<uint8_t>& is_valid, int repeats = 1) {
+  int64_t length = array.length();
+  int64_t base_length = static_cast<int64_t>(strings.size());
+  ASSERT_EQ(base_length, static_cast<int64_t>(is_valid.size()));
+
+  int32_t value_pos = 0;
+  int32_t value_length;
+  for (int i = 0; i < length; ++i) {
+    auto j = i % base_length;
+    if (is_valid[j]) {
+      ASSERT_FALSE(array.IsNull(i));
+      array.GetValue(i, &value_length);
```

Comment:
Might want to get a string_view, could simplify this body a tiny bit, or make it clearer.

Reply:
Hmm, right.